src/main/java/in/ashwanthkumar/gocd/github/provider/gitlab/GitLabProvider.java

```diff
@@ -0,0 +1,155 @@
+        data.put("PR_ID", prId);
+
+        PullRequestStatus prStatus = null;
+        boolean isDisabled = System.getProperty("go.plugin.gitlab.pr.populate-details", "Y").equals("N");
+        LOG.debug("Populating PR details is disabled");
+        if (!isDisabled) {
+            prStatus = getPullRequestStatus(gitConfig, prId, prSHA);
+        }
+
+        if (prStatus != null) {
+            data.put("PR_BRANCH", String.valueOf(prStatus.getPrBranch()));
+            data.put("TARGET_BRANCH", String.valueOf(prStatus.getToBranch()));
+            data.put("PR_URL", String.valueOf(prStatus.getUrl()));
+            data.put("PR_AUTHOR", prStatus.getAuthor());
+            data.put("PR_AUTHOR_EMAIL", prStatus.getAuthorEmail());
+            data.put("PR_DESCRIPTION", prStatus.getDescription());
+            data.put("PR_TITLE", prStatus.getTitle());
+        }
+    }
+
+    @Override
+    public ScmPluginConfigurationView getScmConfigurationView() {
+        return new DefaultScmPluginConfigurationView();
+    }
+
+    @Override
+    public GeneralPluginConfigurationView getGeneralConfigurationView() {
+        return new DefaultGeneralPluginConfigurationView();
+    }
+
+    private PullRequestStatus getPullRequestStatus(GitConfig gitConfig, String prId, String prSHA) {
+        try {
+            MergeRequest currentPR = pullRequestFrom(gitConfig, Integer.parseInt(prId));
+            return transformMergeRequestToPullRequestStatus(prSHA).apply(currentPR);
+        } catch (Exception e) {
+            // ignore
+            LOG.warn(e.getMessage(), e);
+        } return null;
+    }
+
+    private MergeRequest pullRequestFrom(GitConfig gitConfig, int currentPullRequestID) throws GitLabApiException,
+            NullPointerException {
+        return Objects.requireNonNull(loginWith(gitConfig))
+                .getMergeRequestApi()
+                .getMergeRequest(GitLabUtils.parseGitlabUrl(gitConfig.getEffectiveUrl()), currentPullRequestID);
+    }
+
+    private Function<MergeRequest, PullRequestStatus> transformMergeRequestToPullRequestStatus(final String mergedSHA) {
+        return new Function<MergeRequest, PullRequestStatus>() {
+            @Override
+            public PullRequestStatus apply(MergeRequest input) {
+                int prID = input.getId();
+                Author user = input.getAuthor();
+                return new PullRequestStatus(prID, GitLabProvider.REF_PATTERN, input.getSha(), mergedSHA,
+                        input.getSourceBranch(), input.getTargetBranch(), input.getWebUrl(), user.getName(),
+                        user.getEmail(), input.getDescription(), input.getTitle());
+            }
+        };
+    }
+
+    private GitLabApi loginWith(GitConfig gitConfig) throws GitLabApiException {
+        if (hasCredentials(gitConfig))
+            return GitLabApi.oauth2Login(gitConfig.getUrl(), gitConfig.getUsername(), gitConfig.getPassword());
```

Comment:
This seems to be expecting the credentials to be provided as part of every material. Does the API library support reading the credentials from a common path like how the Github API support? Also does the API Token work as Password for Gitlab or do we need pass our actual Password? If that's the case, what happens where folks have in-house Gitlab installation with SSO?

Reply:
`addConfigData` method in GitLab provider will be called while fetching `GitConfig`. This method should ensure that the properties are read before any operation is performed. On the API Token part, the API has support for login using API token.